File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution
{

public:
    int longestConsecutive(vector<int> &nums)
    {
        if(nums.size()==0)
            return 0;
        unordered_set<int> s(nums.begin(), nums.end());
        int longestSofar = 1;
        for (auto num : s)
        {
            if (s.count(num - 1) == 0)
            {
                int conseq = 1;

                while (s.count(++num))
                {
                    conseq++;
                }

                longestSofar = max(longestSofar, conseq);
            }
        }

        return longestSofar;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
import java.util.*;

class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0) {
            return 0; // Return 0 for an empty array
        }

        Set<Integer> set = new HashSet<>();
        for (int num : nums) {
            set.add(num);
        }

        int maxCount = 0;
        for (int num : nums) {
            // Start of a sequence if num - 1 is not in the set
            if (!set.contains(num - 1)) {
                int currentCount = 1;
                int currentNum = num;

                // Count the length of the sequence
                while (set.contains(currentNum + 1)) {
                    currentNum++;
                    currentCount++;
                }

                // Update the maximum length
                maxCount = Math.max(maxCount, currentCount);
            }
        }

        return maxCount;
    }
}
